Mathlib/Computability/DFA.lean

```diff
@@ -163,7 +182,83 @@ theorem pumping_lemma [Fintype σ] {x : List α} (hx : x ∈ M.accepts)
   rw [Set.mem_singleton_iff] at ha' hc'
   substs ha' hc'
   have h := M.evalFrom_of_pow hb hb'
-  rwa [mem_accepts, evalFrom_of_append, evalFrom_of_append, h, hc]
+  rwa [mem_accepts, eval, evalFrom_of_append, evalFrom_of_append, h, hc]
 #align DFA.pumping_lemma DFA.pumping_lemma
 
+section Maps
+
+variable {α' σ' : Type*} (f : α' → α) (g : σ ≃ σ')
+
+/--
+`M.comap f` pulls back the alphabet of `M` along `f`. In other words, it applies `f` to the input
+before passing it to `M`.
+-/
+@[simps]
+def comap : DFA α' σ where
```

Comment:
I am confused by explicit arguments being inserted from `variable`. Afaik this is a matter of style and you can keep it your way.

Reply:
I agree it is quite confusing. I prefer to get exactly the implicit arguments from `variable` statements

Reply:
This is exactly what I would do, but I don't want to impose my style on others so much. I guess if I really care about it, I should make a RFC for a convention to use `variable` only with `{}` and `[]` and possibly a linter for that, which does not belong to this PR. So @lambda-fairy feel free to ignore this comment. At this moment, this style issue is up to you.

Reply:
I don't really mind either way. I've changed it to be more explicit.